dashboard/app/lib/javascripts/dashboard/main.js

```diff
@@ -65,6 +75,91 @@ window.Dashboard = {
 		Marbles.history.navigate('login'+ redirectPath);
 	},
 
+	__catchInsecurePingResponse: function(httpsArgs) {
+		var httpsXhr = httpsArgs[1], self = this,
+			handleSuccess, handleError;
+
+		handleSuccess = function (httpArgs) {
+			var httpXhr = httpArgs[1];
+			// https did not work but http did...something is wrong with the cert
+			self.Dispatcher.handleAppEvent({
+				name: "HTTPS_CERT_MISSING",
+				status: httpXhr.status
+			});
+			return Promise.reject(new Error('HTTPS certificate is not trusted.'));
+		};
+		handleError = function (httpArgs) {
+			var httpXhr = httpArgs[1];
+			if (httpXhr === undefined) {
```

Comment:
The problem is that handleError always get's called if handleSuccess got called, because handleSuccess *always* returns a Promise.reject(): `return self.client.ping("controller", "http").then(handleSuccess).catch(handleError);
`